### PR TITLE
Example showing how to use a master detail page

### LINKED ISSUE #53: [Update] 09 MasterDetail
### Sample

09-MasterDetail

#### Reason for Update

This sample is currently stubbed and needs actual implementation.

TODO 

- All platforms should show a Hamburger icon
- Show use with a ContentPage
- Show use with a TabbedPage

### PR DESCRIPTION
**Pull request by ScarlettCode**
closes #53 

This example shows

- How to use the Master Detail Page
- Conventional use with a NavigationPage / Content Page
- Less common use with a Tabbed Page as the Child
- Hamburger icon on all platforms

I'm not able to test that it works on iOS and shows the hamburger menu there. Holding thumbs it just works 🤷‍♂️. Working on Android and UWP

### CODE REVIEW

File: 09-MasterDetail/src/PrismSample/Views/HomePage.xaml [hunk 1]
```diff
@@ -0,0 +1,20 @@
+﻿<?xml version="1.0" encoding="utf-8" ?>
+<MasterDetailPage xmlns="http://xamarin.com/schemas/2014/forms"
+                  xmlns:x="http://schemas.microsoft.com/winfx/2009/xaml"
+                  xmlns:prism="http://prismlibrary.com"
+                  prism:ViewModelLocator.AutowireViewModel="True"
+                  x:Class="PrismSample.Views.HomePage" 
+                  MasterBehavior="Popover">
+
+    <MasterDetailPage.Master>
+        <ContentPage Title="Menu">
```
Comment (dansiegel) [inline 09-MasterDetail/src/PrismSample/Views/HomePage.xaml]: you need to specify an Icon for iOS

File: 09-MasterDetail/src/PrismSample/Views/HomePage.xaml [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+            <StackLayout Padding="20">
+
```
Comment (dansiegel) [inline 09-MasterDetail/src/PrismSample/Views/HomePage.xaml]: also be sure to clear trailing whitespace or whitespace on empty lines

File: 09-MasterDetail/src/PrismSample/Views/TabsPage.xaml [hunk 3]
```diff
@@ -0,0 +1,15 @@
...
+            prism:ViewModelLocator.AutowireViewModel="True"
+            x:Class="PrismSample.Views.TabsPage"
+            Title="Tabbed Page">
+  
+  <TabbedPage.Children>
+    <views:TabAPage Title="Tab A" />
+    <views:TabBPage Title="Tab B"/>
+  </TabbedPage.Children>
+
+</TabbedPage>
```
Comment (dansiegel) [inline 09-MasterDetail/src/PrismSample/Views/TabsPage.xaml]: just use a TabbedPage and create this on the fly

File: 09-MasterDetail/src/PrismSample/Views/HomePage.xaml [hunk 4]
```diff
@@ -0,0 +1,20 @@
+﻿<?xml version="1.0" encoding="utf-8" ?>
+<MasterDetailPage xmlns="http://xamarin.com/schemas/2014/forms"
+                  xmlns:x="http://schemas.microsoft.com/winfx/2009/xaml"
+                  xmlns:prism="http://prismlibrary.com"
+                  x:Class="PrismSample.Views.HomePage" 
+                  MasterBehavior="Popover"
+                  IconImageSource="hamburger.png">
```
Comment (dansiegel) [inline 09-MasterDetail/src/PrismSample/Views/HomePage.xaml]: this is in the wrong place as it should be on the Master not the MasterDetailPage

File: 09-MasterDetail/src/PrismSample/Views/HomePage.xaml (hunk 1 above)
Comment (dansiegel) [reply 09-MasterDetail/src/PrismSample/Views/HomePage.xaml]: see above

File: 09-MasterDetail/src/PrismSample/Views/HomePage.xaml.cs [hunk 5]
```diff
@@ -0,0 +1,12 @@
...
+
+namespace PrismSample.Views
+{
+    public partial class HomePage : MasterDetailPage
+    {
+        public HomePage()
+        {
+            InitializeComponent();
+        }
+    }
```
Comment (dansiegel) [inline 09-MasterDetail/src/PrismSample/Views/HomePage.xaml.cs]: Please implement `IMasterDetailPageOptions`

File: 09-MasterDetail/src/PrismSample/Views/MainPage.xaml [hunk 6]
```diff
@@ -7,10 +7,8 @@
              Title="Main Page"
              x:Class="PrismSample.Views.MainPage">
 
-    <StackLayout>
-        <Label Text="Welcome to Xamarin.Forms!" 
-           HorizontalOptions="Center"
-           VerticalOptions="CenterAndExpand" />
-    </StackLayout>
+  <StackLayout>
+    <Label Text="Welcome to Xamarin.Forms!" HorizontalOptions="Center" VerticalOptions="CenterAndExpand" />
```
Comment (dansiegel) [inline 09-MasterDetail/src/PrismSample/Views/MainPage.xaml]: Change the Text here... ??? "Welcome to MasterDetailPage with Prism for Xamarin.Forms!"

File: 09-MasterDetail/src/PrismSample/ViewModels/TabAPageViewModel.cs [hunk 7]
```diff
@@ -0,0 +1,12 @@
...
+namespace PrismSample.ViewModels
+{
+    public class TabAPageViewModel : BindableBase
+    {
+        public TabAPageViewModel()
+        {
+
+        }
+    }
+}
```
Comment (dansiegel) [inline 09-MasterDetail/src/PrismSample/ViewModels/TabAPageViewModel.cs]: Implement IInitialize... and pull from the NavigationParameters a message for this tab...

**Comment by ChaseFlorell** [comment]
I am definitely learning a lot just reading this and reading Dan's comments. We have a MDP implementation in our backlog right now and this is exactly what we need!

**Pull request closed without merging**